Runtime/Model/Breadcrumbs/InMemory/BacktraceInMemoryLogManager.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Backtrace.Unity.Common;
+using System.Collections.Generic;
+
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("Backtrace.Unity.Tests.Runtime")]
+namespace Backtrace.Unity.Model.Breadcrumbs.InMemory
+{
+    internal sealed class BacktraceInMemoryLogManager : IBacktraceLogManager
+    {
+        /// <summary>
+        /// Default maximum number of in memory breadcrumbs
+        /// </summary>
+        public const int DefaultMaximumNumberOfInMemoryBreadcrumbs = 100;
+
+        /// <summary>
+        /// Maximum number of in memory breadcrumbs
+        /// </summary>
+        public int MaximumNumberOfBreadcrumbs { get; set; } = DefaultMaximumNumberOfInMemoryBreadcrumbs;
+
+        /// <summary>
+        /// Lock object
+        /// </summary>
+        private object _lockObject = new object();
+
+        /// <summary>
+        /// Breadcrumbs
+        /// </summary>
+        internal readonly Queue<InMemoryBreadcrumb> Breadcrumbs = new Queue<InMemoryBreadcrumb>(DefaultMaximumNumberOfInMemoryBreadcrumbs);
+
+        private long _breadcrumbId = DateTimeHelper.Timestamp();
+
+        /// <summary>
+        /// Returns path to breadcrumb file - which is string.Empty for in memory breadcrumb manager
+        /// </summary>
+        public string BreadcrumbsFilePath
+        {
+            get
+            {
+                return string.Empty;
+            }
+        }
+
+        public bool Add(string message, BreadcrumbLevel type, UnityEngineLogLevel level, IDictionary<string, string> attributes)
+        {
+            lock (_lockObject)
```

Comment:
Do you need to lock this? Can you use a ConcurrentQueue to avoid this lock?

Reply:
We can't use ConcurrentQueue because this data structure is available in .NET 4.0+. We still support .NET 3.5.